guide/src/free-threading.md

```diff
@@ -0,0 +1,165 @@
+# Supporting Free-Threaded CPython
+
+CPython 3.13 introduces an experimental "free-threaded" build of CPython that
+does not rely on the [global interpreter
+lock](https://docs.python.org/3/glossary.html#term-global-interpreter-lock)
+(often referred to as the GIL) for thread safety. As of version 0.23, PyO3 also
+has preliminary support for building Rust extensions for the free-threaded
+Python build and support for calling into free-threaded Python from Rust.
+
+If you want more background on free-threaded Python in general, see the [what's
+new](https://docs.python.org/3.13/whatsnew/3.13.html#whatsnew313-free-threaded-cpython)
+entry in the CPython docs, the [HOWTO
+guide](https://docs.python.org/3.13/howto/free-threading-extensions.html#freethreading-extensions-howto)
+for porting C extensions, and [PEP 703](https://peps.python.org/pep-0703/),
+which provides the technical background for the free-threading implementation in
+CPython.
+
+The main benefit for supporting free-threaded Python is that both pure Python
+code and code interacting with Python via the CPython C API can run
+simultaneously on multiple OS threads.  This means it's much more
+straightforward to achieve multithreaded parallelism in the Python layer, using
+the `threading` module, without any need to ensure the GIL is released during
+compute-heavy tasks, or any worries about [Amdahl's
+law](https://en.wikipedia.org/wiki/Amdahl%27s_law) limiting parallel
+performance. If you have ever needed to use `multiprocessing` to achieve a
```

Comment:
I don't understand the amdahl's law reference here, it's absolutely still relevant :-)

I think what you're trying to say is "in a GIL world, the value of releasing the GIL is bound by the portion of your code that drops the GIL (under Amdahl's law), while in the no GIL world, you can gain teh benefits to the full extent your code is parallel", but that's not clear from the text.

Reply:
I rewrote this to try to make the point about Amdahl's law more explicit. Thanks for pointing out that this was unclear.

Reply:
Looks good